### ISSUE TITLE
BadWindow error crashes Python script

### ISSUE THREAD

**Issue by Brimill** [intel-isl/Open3D#3443]
**Description**
When using a Visualizer object to call create_window(), closing it and calling create_window() again, the Python script crashes with the following error:

```
X Error of failed request:  BadWindow (invalid Window parameter)
Major opcode of failed request:  18 (X_ChangeProperty)
Resource id in failed request:  0x0
Serial number of failed request:  244
Current serial number in output stream:  251
```
This may be related  to #1715, due to the similarity in the steps to reproduce the issue. However, the error message is slightly different.

**To Reproduce**

```
import open3d as o3d

vis = o3d.visualization.Visualizer()
vis.create_window()
vis.run()
vis.destroy_window()
vis.create_window() # This is the line the error occurs
vis.run()
vis.destroy_window()
vis.destroy()
```

**Environment:**

- Operating system: Ubuntu 20.10
- Python version: Python 3.8.6
- Open3D version: 0.12.0
- Is this remote workstation?: no
- How did you install Open3D?: pip

**Comment by okrupa** [comment]
hello I have the same problem. Did you manage to solve it?

**Comment by Brimill** [comment]
No, I haven't. It has been a while but if I recall correctly I just worked around the issue.
Maybe what we are doing is not how the Visualizer object is intended to be used.

**Comment by domef** [comment]
I have the same problem

**Comment by HowcanoeWang** [comment]
The same problem on the WSL2 of Ubuntu 20.04
